### PR TITLE
Documentation for resource views

### PR DESCRIPTION
**Pull request by amercader**
- [x] Overview
- [x] How to enable them
- [x] How to define the default ones
- [x] Core plugins
- [x] Migrating from previous CKAN versions
- [ ] Writing view plugins

### CODE REVIEW

**Comment by amercader** [comment]
@seanh are you up for reading some entertaining and informative docs?

Rendered here:

http://ckan.readthedocs.org/en/2205-resource-views-docs/maintaining/data-viewer.html
http://ckan.readthedocs.org/en/2205-resource-views-docs/extensions/plugin-interfaces.html#ckan.plugins.interfaces.IResourceView

**Comment by seanh** [comment]
@amercader Sure, I'll take a look now

File: doc/maintaining/data-viewer.rst [hunk 1]
```diff
@@ -1,223 +1,332 @@
...
-resource being viewed.
-In general, images will be directly embedded, unstructured or plain text
-files will be loaded in an iframe, and more complex data types will need to
-use a custom widget.
+Overview
+--------
 
-The data preview functionality that is provided by CKAN is described in
-the following sections:
+The CKAN resource page can contain one or more views of the resource data
```
Comment (seanh) [inline doc/maintaining/data-viewer.rst]: I would say "one or more visualizations" then put examples in brackets (a bar chart, a table, etc). Then go on to say "These are commonly referred to as *resource views*". I think you have to explain it without using the term *resource view* first, then once you've introduced the term you can use it from then on. The reason is that *resource view* is a CKAN jargon term, will not mean anything to people at first.

File: doc/maintaining/data-viewer.rst [hunk 2]
```diff
@@ -1,223 +1,332 @@
...
+  and some graphs for tabular data).
 
-.. _image-preview:
+* Dataset editors can choose which views to show, reorder them and configure
+  them individually.
 
-Viewing images and text files
------------------------------
+* Views configuration persits on the database, so it's not lost on each page
+  refresh.
```
Comment (seanh) [inline doc/maintaining/data-viewer.rst]: I might just delete this. Of course it persists, that's what people would assume. Of course, it didn't used to persist. But users don't need to know about that.

(The right place for pointing out "it didn't used to persist but now it does" would be in places like changelog, release notes or announcement, feature announcement blog post.)

File: doc/maintaining/data-viewer.rst [hunk 3]
```diff
@@ -1,223 +1,332 @@
...
-CKAN config file. See `ckan.preview.loadable` for more information.
+.. image:: /images/manage_views.png
 
-Note that these documents will be directly linked by the browser, so the
-way in which they are shown may vary. If you want to ensure for instance that
-XML based documents are correctly previewed, have a look at `Viewing highlighted XML, JSON and plain text data`_.
 
+The *New view* dropdown will show the available view types for this particular
+resource. If the list if empty, you may need to add the relevant view plugins
+to the :ref:`ckan.plugins` setting on your configuration file, eg::
```
Comment (seanh) [inline doc/maintaining/data-viewer.rst]: Typo: If the list if empty,

File: ckan/plugins/interfaces.py [hunk 4]
```diff
@@ -206,78 +206,146 @@ def notify(self, resource):
...
-        :param preview_enabled:
-            Says if the preview button appears for this resource. Some preview
-            types have their  previews integrated with the form.
-            Some preview types have their previews integrated with the form.
-            Default false (Optional)
-        :param full_page_edit:  Says if the edit form is the full page width
-            of the page. Default false (Optional)
-
-        eg:
+        Returns a dictionarty with configuration options for the view.
```
Comment (seanh) [inline ckan/plugins/interfaces.py]: Typo

**Comment by seanh** [comment]
When listing all the default view types, a screenshot of each would be great.

All minor suggestions, I think these docs are great and could be merged as-is

**Comment by amercader** [comment]
@seanh added fixes and screenshots, feel free to merge if happy

**Pull request merged**